src/tutorial/counter-app/ratatui-counter-app/src/app.rs

```diff
@@ -48,3 +47,21 @@ impl App {
   }
 }
 ///// ANCHOR_END: application_impl
+
+///// ANCHOR: application_test
+mod tests {
+  #[test]
+  fn test_app_increment_counter() {
+    let mut app = super::App::default();
+    app.increment_counter();
+    assert_eq!(app.counter, 1);
+  }
+
+  #[test]
+  fn test_app_decrement_counter() {
+    let mut app = super::App::default();
+    app.decrement_counter();
+    assert_eq!(app.counter, 0);
+  }
+}
```

Comment:
```suggestion
mod tests {
  use super::*;
  #[test]
  fn test_app_increment_counter() {
    let mut app = App::default();
    app.increment_counter();
    assert_eq!(app.counter, 1);
  }

  #[test]
  fn test_app_decrement_counter() {
    let mut app = App::default();
    app.decrement_counter();
    assert_eq!(app.counter, 0);
  }
}
```

Reply:
This causes the counting-app to not compile (which is why I used `super` in line previously).
Taking the compiler's suggestion to do `use crate::app::App;` also results in a failure to compile. Maybe I'm doing something wrong?
```
 Compiling counting-app v0.1.0 (/home/raffaele/RustProjects/ratatui/counting-app)
warning: `counting-app` (lib) generated 1 warning (run `cargo fix --lib -p counting-app` to apply 1 suggestion)
error[E0433]: failed to resolve: use of undeclared type `App`
  --> src/app.rs:40:23
   |
40 |         let mut app = App::default();
   |                       ^^^ use of undeclared type `App`
   |
help: consider importing this struct
   |
38 +     use crate::app::App;
   |

error[E0433]: failed to resolve: use of undeclared type `App`
  --> src/app.rs:46:23
   |
46 |         let mut app = App::default();
   |                       ^^^ use of undeclared type `App`
   |
help: consider importing this struct
   |
38 +     use crate::app::App;
   |

For more information about this error, try `rustc --explain E0433`.
warning: `counting-app` (lib test) generated 1 warning (1 duplicate)
error: could not compile `counting-app` (lib test) due to 2 previous errors; 1 warning emitted
```

Reply:
Done